spanner/spanner_snippets/snippet.go

```diff
@@ -906,6 +916,264 @@ func queryWithTimestamp(ctx context.Context, w io.Writer, client *spanner.Client
+
+func writeAndReadUsingDML(ctx context.Context, w io.Writer, client *spanner.Client) error {
+	_, err := client.ReadWriteTransaction(
+		ctx, func(ctx context.Context, txn *spanner.ReadWriteTransaction) error {
+			// Insert Record
+			rowCount, err := txn.Update(ctx, spanner.Statement{
+				SQL: `INSERT Singers (SingerId, FirstName, LastName)
+								VALUES (11, 'Timothy', 'Campbell')`})
+			if err != nil {
+				return err
+			}
+			fmt.Fprintf(w, "%d record(s) inserted.\n", rowCount)
+
+			// Read newly inserted record
+			iter := txn.Query(ctx, spanner.Statement{
+				SQL: `SELECT FirstName, LastName FROM Singers WHERE SingerId = 11`})
+			defer iter.Stop()
+
+			for {
+				row, err := iter.Next()
+				if err == iterator.Done || err != nil {
+					break
+				}
+				var firstName, lastName string
+				if err := row.ColumnByName("FirstName", &firstName); err != nil {
+					return err
+				}
+				if err := row.ColumnByName("LastName", &lastName); err != nil {
+					return err
+				}
+				fmt.Fprintf(w, "Found record name with %s, %s", firstName, lastName)
+			}
+			return err
+		})
+	return err
+}
+
+// [END spanner_dml_write_then_read]
+
+// [START spanner_dml_structs]
+
+func updateUsingDMLStruct(ctx context.Context, w io.Writer, client *spanner.Client) error {
+	_, err := client.ReadWriteTransaction(
+		ctx, func(ctx context.Context, txn *spanner.ReadWriteTransaction) error {
+
+			type nameStruct struct {
+				FirstName string
+				LastName  string
+			}
+			var singerInfo = nameStruct{"Timothy", "Campbell"}
+
+			rowCount, err := txn.Update(ctx, spanner.Statement{
+				SQL: `Update Singers Set LastName = 'Grant' 
+						WHERE STRUCT<FirstName String, LastName String>(Firstname, LastName) = @name`,
+				Params: map[string]interface{}{"name": singerInfo},
+			})
+			if err != nil {
+				return err
+			}
+			fmt.Fprintf(w, "%d record(s) inserted.\n", rowCount)
+			return nil
+		})
+	return err
+}
+
+// [END spanner_dml_structs]
+
+// [START spanner_dml_getting_started_insert]
+
+func writeUsingDML(ctx context.Context, w io.Writer, client *spanner.Client) error {
+	_, err := client.ReadWriteTransaction(
+		ctx, func(ctx context.Context, txn *spanner.ReadWriteTransaction) error {
+			rowCount, err := txn.Update(ctx, spanner.Statement{
+				SQL: `INSERT Singers (SingerId, FirstName, LastName) VALUES
+								(12, 'Melissa', 'Garcia'),
+								(13, 'Russell', 'Morales'),
+								(14, 'Jacqueline', 'Long'),
+								(15, 'Dylan', 'Shaw')`})
+			if err != nil {
+				return err
+			}
+			fmt.Fprintf(w, "%d record(s) inserted.\n", rowCount)
+			return err
+		})
+	return err
+}
+
+// [END spanner_dml_getting_started_insert]
+
+// [START spanner_dml_getting_started_update]
+
+func writeWithTransactionUsingDML(ctx context.Context, w io.Writer, client *spanner.Client) error {
+	_, err := client.ReadWriteTransaction(
+		ctx, func(ctx context.Context, txn *spanner.ReadWriteTransaction) error {
+			// Helper function for retrieving the budget of a record
```

Comment:
I wonder if this comment should start with the thing being described (`getBudget`)? May or may not be more clear. Same with `updateBudget`.

Side note: the "get" prefix is usually avoided in Go, in favor of naming directly after the thing (`foo` instead of `getFoo`), but I think this is OK. See https://golang.org/doc/effective_go.html#Getters.